File: Sorting/Insertion.cpp
#include<iostream>
#include<vector>
using namespace std;

void InsertionSort(vector<int> & arr){
    for(int i = 1; i<arr.size(); i++){
        int curr = arr[i];
        int j = i - 1;
        while(j >= 0 && curr < arr[j]){
            arr[j+1] = arr[j];
            j--; 
        }
        arr[j+1] = curr;
    }
}

int main(){
    vector<int> arr = {2,4,1,7,3};
    InsertionSort(arr);
    for(int ele : arr){
        cout<<ele<<", ";
    }
}

File: Sorting/Insertion.py
def InsertionSort(arr):
    for i in range(1,len(arr)):
        j = i - 1
        curr = arr[i]
        while j >= 0 and curr < arr[j]:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = curr

arr = [2,4,1,5,6]
print(arr)
InsertionSort(arr)
print(arr)